webcompat/views.py

```diff
@@ -169,77 +171,93 @@ def show_issues():
 
 @app.route('/issues/new', methods=['GET', 'POST'])
 def create_issue():
-    """Create a new issue.
-
-    GET will return an HTML response for reporting issues.
-    POST will create a new issue.
+    """Create a new issue or prefill a form for submission.
+
+    * HTTP GET with (optional) parameters
+      * create a form with prefiled data.
+      * parameters:
+        * url: URL of the Web site
+        * src: source of the request (web, addon, etc.)
+        * label: controled list of labels
+    * HTTP POST with a JSON payload
+      * create a form with prefiled data
+      * content-type is application/json
+      * json may include:
+        * title
+        * User agent string
+        * OS identification
+        * labels list
+        * type of bugs
+        * short summary
+        * full description
+        * tested in another browser
+        * body
+    * HTTP POST with an attached form
+      * submit a form to GitHub to create a new issue
+      * form submit type:
+        * authenticated: Github authentification
+        * anonymous: handled by webcompat-bot
 
     Any deceptive requests will be ended as a 400.
     See https://tools.ietf.org/html/rfc7231#section-6.5.1
     """
     # Starting a logger
     log = app.logger
     log.setLevel(logging.INFO)
-    # Get the User-Agent
-    user_agent = request.headers.get('User-Agent')
-    # GET Requests
-    if request.method == 'GET':
-        bug_form = get_form(user_agent)
-        if g.user:
-            get_user_info()
-        # Note: `src` and `label` are special GET params that can pass
-        # in extra information about a bug report. They're not part of the
-        # HTML <form>, so we stick them in the session cookie so they survive
-        # the scenario where the user decides to do authentication, and they
-        # can then be passed on to form.py
-        if request.args.get('src'):
-            session['src'] = request.args.get('src')
-        if request.args.get('label'):
-            session['label'] = request.args.getlist('label')
+    if g.user:
+        get_user_info()
+    # We define which type of requests we are dealing with.
+    request_type = form_type(request)
+    # Form Prefill section
+    if request_type == 'prefill':
+        form_data = prepare_form(request)
+        bug_form = get_form(form_data)
+        session['form_data'] = form_data
```

Comment:
...ok, understanding some things a little bit more. I think 😄 

 `session.form_data` is only used to pass along the `extra_labels` inside the `request_type == 'create'` block. So do we really need this copy of `form_data` in the session at all? We could either pass `extra_labels` in the `request.form` itself, via a `HiddenInput`, or just toss it in a session key without the rest of the form.

Reply:
so yes initially there was form and form_data and it was a bit confusing. `form` in the PR is the form object. `form_data` is a dictionary. Initially I created the dictionary for saving things in between sessions, like you did for a couple of other things. And my train of thoughts let's preserve what we need in case we need it later.

I tried to use an `HiddenInput` for `extra_labels`, but it becomes extra_dirty_labels :) aka the dictionary gets modified as a string because it is what a form can do. 

I don't think there is a way to store a `HiddenArrayInput` except by making a loop over all values of `extra_labels` and adding an individual `HiddenInput` for each of them.
**We could do that** What do you think? session or state management through values. 

We could also do a stringify json, and reparse it after. 

Many possibilities. 💭

Reply:
Thanks. I think all the special casing for extra_labels as a hidden input is probably not worth it, at least right now. We can keep it as-is with `form_data` in session, or only store and re-store the `extra_labels` value in session. Whichever you think is better.

Reply:
I changed back to extra_labels and we can reuse form_data another time if needed.